ExoPlayerIMAAdRulesSampleApp/src/main/java/com/brightcove/player/samples/exoplayer/ima/adrules/MainActivity.java

```diff
@@ -0,0 +1,159 @@
+package com.brightcove.player.samples.exoplayer.ima.adrules;
+
+import android.os.Bundle;
+import android.util.Log;
+import com.brightcove.ima.GoogleIMAComponent;
+import com.brightcove.ima.GoogleIMAEventType;
+import com.brightcove.ima.GoogleIMAVideoAdPlayer;
+import com.brightcove.player.event.Event;
+import com.brightcove.player.event.EventEmitter;
+import com.brightcove.player.event.EventListener;
+import com.brightcove.player.event.EventType;
+import com.brightcove.player.media.Catalog;
+import com.brightcove.player.media.VideoFields;
+import com.brightcove.player.media.VideoListener;
+import com.brightcove.player.model.Video;
+import com.brightcove.player.view.BrightcovePlayer;
+import com.brightcove.player.view.ExoPlayerVideoView;
+import com.brightcove.player.util.StringUtil;
+import com.google.ads.interactivemedia.v3.api.AdDisplayContainer;
+import com.google.ads.interactivemedia.v3.api.AdsRequest;
+import com.google.ads.interactivemedia.v3.api.ImaSdkFactory;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * This app illustrates how to use "Ad Rules" the Google IMA plugin
+ * and the Brightcove Player for Android.  Note: cue points are not
+ * used with Ad Rules.
+ *
+ * @author Paul Matthew Reilly (original code)
```

Comment:
Done. Thanks for this reminder.
I've also updated attributions to other projects that I've adapted from existing code.